File: lib/models/order_model.dart
import 'package:bitbybit/models/schedule_model.dart';
import 'package:bitbybit/widgets/weekindicator.dart';
import 'package:cloud_firestore/cloud_firestore.dart';

class OrderItem {
  bool active;
  double amount;
  Timestamp createdTimestamp;
  Timestamp updatedTimestamp;
  String exchange;
  String pair;
  String documentId;
  Schedule schedule;

  OrderItem(
      {this.active,
      this.amount,
      this.createdTimestamp,
      this.updatedTimestamp,
      this.exchange,
      this.pair,
      this.schedule});

  OrderItem.fromJson(Map<String, dynamic> json) {
    active = json['active'];
    amount = json['amount'];
    createdTimestamp = json['createdTimestamp'];
    updatedTimestamp = json['updatedTimestamp'];
    exchange = json['exchange'];
    pair = json['pair'];
    if(json["schedule"]!=null) {
      schedule = Schedule(
          monday: json['schedule']['monday'] ?? false,
          tuesday: json['schedule']['tuesday'] ?? false,
          wednesday: json['schedule']['wednesday'] ?? false,
          thursday: json['schedule']['thursday'] ?? false,
          friday: json['schedule']['friday'] ?? false,
          saturday: json['schedule']['saturday'] ?? false,
          sunday: json['schedule']['sunday'] ?? false);
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['active'] = this.active;
    data['amount'] = this.amount;
    data['createdTimestamp'] = this.createdTimestamp;
    data['updatedTimestamp'] = this.updatedTimestamp;
    data['exchange'] = this.exchange;
    data['pair'] = this.pair;
    data['schedule'] = this.schedule.toJson();
    return data;
  }
}

File: Backend/functions/src/models/order_model.ts
export class Schedule {
    private _monday: boolean = false;
    private _tuesday: boolean = false;
    private _wednesday: boolean = false;
    private _thursday: boolean = false;
    private _friday: boolean = false;
    private _saturday: boolean = false;
    private _sunday: boolean = false;

    get monday(): boolean {
        return this._monday;
    }

    set monday(value: boolean) {
        this._monday = value;
    }

    get tuesday(): boolean {
        return this._tuesday;
    }

    set tuesday(value: boolean) {
        this._tuesday = value;
    }

    get wednesday(): boolean {
        return this._wednesday;
    }

    set wednesday(value: boolean) {
        this._wednesday = value;
    }

    checkValidity(x): boolean {
        if (x == null) {
            return true;
        }

        if (x === null) {
            return true;
        }

        if (typeof x === "undefined") {
            return true;
        }
        return false;
    }

    get thursday(): boolean {
        /*if (this.checkValidity(this._thursday)) {
            return false;
        }*/
        return this._thursday;
    }

    set thursday(value: boolean) {
        this._thursday = value;
    }

    get friday(): boolean {
        return this._friday;
    }

    set friday(value: boolean) {
        this._friday = value;
    }

    get saturday(): boolean {
        return this._saturday;
    }

    set saturday(value: boolean) {
        this._saturday = value;
    }

    get sunday(): boolean {
        return this._sunday;
    }

    set sunday(value: boolean) {
        this._sunday = value;
    }
}

export class Order {
    private _exchange: string = "";
    private _amount: number = 0;
    private _pair: string = "";
    private _active: boolean = false;
    private _userId: string = "";
    private _schedule: Schedule = new Schedule();

    get schedule(): Schedule {
        return this._schedule;
    }

    set schedule(value: Schedule) {
        this._schedule = value;
    }

    get userId(): string {
        return this._userId;
    }

    set userId(value: string) {
        this._userId = value;
    }

    get active(): boolean {
        return this._active;
    }

    set active(value: boolean) {
        this._active = value;
    }

    constructor() {
    }

    get exchange(): string {
        return this._exchange;
    }

    set exchange(value: string) {
        this._exchange = value;
    }

    get amount(): number {
        return this._amount;
    }

    set amount(value: number) {
        this._amount = value;
    }

    get pair(): string {
        return this._pair;
    }

    set pair(value: string) {
        this._pair = value;
    }
}